src/main/java/com/atypon/wayf/verticle/routing/DeviceRoutingProvider.java

```diff
@@ -173,31 +173,33 @@ public Completable registerLocalId(RoutingContext routingContext) {
                     String hashedLocalId = deviceFacade.encryptLocalId(publisher.getId(), localId);
 
                     return deviceFacade.relateLocalIdToDevice(publisher, hashedLocalId)
-                            .map((device) -> {
-                                String globalId = device.getGlobalId();
+                            .map((device) -> addGlobalIdCookie(device, routingContext));
 
-                                Cookie cookie = new CookieImpl(RequestReader.DEVICE_ID, globalId)
-                                        .setDomain(wayfDomain)
-                                        .setMaxAge(158132000l)
-                                        .setPath("/");
+                });
+    }
+
+    private Device addGlobalIdCookie(Device device, RoutingContext rc) {
+        String globalId = device.getGlobalId();
 
-                                String requestOrigin = RequestReader.getHeaderValue(routingContext, "Origin");
+        Cookie cookie = new CookieImpl(RequestReader.DEVICE_ID, globalId)
+                .setDomain(wayfDomain)
+                .setMaxAge(158132000l)
+                .setPath("/");
 
-                                LOG.debug("Request origin [{}]", requestOrigin);
+        String requestOrigin = RequestReader.getHeaderValue(rc, "Origin");
 
-                                if (requestOrigin == null || requestOrigin.isEmpty()) {
-                                    throw new ServiceException(HttpStatus.SC_BAD_REQUEST, "Origin header is required");
-                                }
+        LOG.debug("Request origin [{}]", requestOrigin);
 
-                                routingContext.response().putHeader("Access-Control-Allow-Origin", requestOrigin);
+        if (requestOrigin == null || requestOrigin.isEmpty()) {
```

Comment:
I don't think we need to throw an error here. just don't send the Access-control-allow-origin header if the request Origin header is missing